File: basic/basic_algorithm/hello coding 그림으로 개념을 이해하는 알고리즘/퀵정렬.js
/**
 * 퀵 정렬 ** 수정 필요~!
 */

function quickSort(array) {
  if (array.length < 2) {
    return array;
  } else {
    let pivot = array[0];
    let less = [],
      greater = [];
    for (let i = 1; i < array.length; i++) {
      if (i <= pivot) {
        less.push(array[i]);
      } else {
        greater.push(array[i]);
      }
      console.log(quickSort(less), [pivot], quickSort(greater));
      return quickSort(less), [pivot], quickSort(greater);
    }
  }
}
console.log(quickSort([2, 4, 6, 3, 1]));

File: basic/basic_algorithm/hello coding 그림으로 개념을 이해하는 알고리즘/퀵정렬.py
# 퀵 정렬


def quickSort(array):
    if len(array) < 2:
        return array
    else:
        pivot = array[0]
        less = [i for i in array[1:] if i <= pivot]
        greater = [i for i in array[1:] if i > pivot]

        return quickSort(less) + [pivot] + quickSort(greater)


print(quickSort([2, 4, 6, 3, 1]))
